airflow/upgrade/rules/legacy_ui_deprecated.py

```diff
@@ -0,0 +1,32 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import absolute_import
+
+from airflow.configuration import conf
+from airflow.upgrade.rules.base_rule import BaseRule
+
+
+class LegacyUIDeprecatedRule(BaseRule):
+    title = "RBAC is enabled by default"
+
+    description = "Legacy UI is deprecated. FAB RBAC is enabled by default in order to increase security."
+
+    def check(self):
+        rbac_conf = conf.get(section="webserver", key="rbac")
+        if not rbac_conf:
```

Comment:
By default in 1.10, we have RBAC disabled. All users must migrate to RBAC prior to migration, so this option must be True (`conf.getboolean(section="webserver", key="rbac") == True`)
See: https://github.com/apache/airflow/blob/master/UPDATING.md#drop-legacy-ui-in-favor-of-fab-rbac-ui

> Before upgrading to this release, we recommend activating the new FAB RBAC UI. For that, you should set the rbac options in [webserver] in the airflow.cfg file to true

Reply:
@mik-laj point taken, thanks. but then shouldn't we be checking if the user has set to **false**, and display the message if it's the case? If the user has rbac = true, we shouldn't display any message, should we?

Reply:
Ideally you should use the same code as the actual webserver command.
https://github.com/apache/airflow/blob/90e984f90b8384cc7a461d8340a48b95af272946/airflow/bin/cli.py#L1124
https://github.com/apache/airflow/blob/90e984f90b8384cc7a461d8340a48b95af272946/airflow/settings.py#L44